### PR TITLE
Switch `users#archived_at` for `users#is_archived`

### PR DESCRIPTION
**Pull request by tmfrnz**
After we merged #108 we discovered that, while it passed specs, it did
not work in some real world use cases.

Instead of having `archived_at`, we're going to use `is_archived` which
can be simply set and we can find out when it happened based on the
`versions` table if we need to.

To ensure that this attribute can only be set when updating the record,
we've moved the logic from `permitted_attributes` to the specific
`permitted_attributes_for_update` method that Pundit uses.

### CODE REVIEW

**Comment by tmfrnz** [comment]
deployed "simpler check" to test site, however it still does not appear to work. 

here is the payload of the PUT request
```
archived_at: null
created_at: "2022-03-14T18:45:09+01:00"
email: "test@test.de"
is_archived: "true"
name: "test"
updated_at: "2022-05-24T16:10:15+02:00"
updated_by_id: 5
```
and here the server response
```
       "id": "5",
        "type": "users",
        "attributes": {
            "created_at": "2022-03-14T18:45:09+01:00",
            "updated_at": "2022-05-24T16:10:15+02:00",
            "updated_by_id": 5,
            "email": "test@test.de",
            "name": "test",
            "archived_at": null,
            "is_archived": false
        }
```

finally here the server log 
```
{"created_at"=>"2022-03-14T18:45:09+01:00", "updated_at"=>"2022-05-24T16:10:15+02:00", "updated_by_id"=>5, "email"=>"test@test.de", "name"=>"test", "archived_at"=>nil, "is_archived"=>"true", "id"=>"5", "user"=>{"email"=>"test@test.de", "created_at"=>"2022-03-14T18:45:09+01:00", "updated_at"=>"2022-05-24T16:10:15+02:00", "name"=>"test", "updated_by_id"=>5, "archived_at"=>nil}}
```

**Comment by tmfrnz** [comment]
@parndt 

deployed "simpler check" to test site, however it still does not appear to work. 

here is the payload of the PUT request
```
archived_at: null
created_at: "2022-03-14T18:45:09+01:00"
email: "test@test.de"
is_archived: "true"
name: "test"
updated_at: "2022-05-24T16:10:15+02:00"
updated_by_id: 5
```
and here the server response
```
       "id": "5",
        "type": "users",
        "attributes": {
            "created_at": "2022-03-14T18:45:09+01:00",
            "updated_at": "2022-05-24T16:10:15+02:00",
            "updated_by_id": 5,
            "email": "test@test.de",
            "name": "test",
            "archived_at": null,
            "is_archived": false
        }
```

finally here the server log 
```
{"created_at"=>"2022-03-14T18:45:09+01:00", "updated_at"=>"2022-05-24T16:10:15+02:00", "updated_by_id"=>5, "email"=>"test@test.de", "name"=>"test", "archived_at"=>nil, "is_archived"=>"true", "id"=>"5", "user"=>{"email"=>"test@test.de", "created_at"=>"2022-03-14T18:45:09+01:00", "updated_at"=>"2022-05-24T16:10:15+02:00", "name"=>"test", "updated_by_id"=>5, "archived_at"=>nil}}
```

**Comment by tmfrnz** [comment]
@parndt maybe you have another idea you want to test but before wasting too much time, let's just get rid of the `archived_at` field and add `is_archived`

**Review by parndt — APPROVED** [review]

File: db/schema.rb [hunk 1]
```diff
@@ -132,6 +132,16 @@
...
+  create_table "feedbacks", force: :cascade do |t|
+    t.text "subject"
+    t.text "content"
+    t.bigint "user_id", null: false
+    t.datetime "notified_at"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["user_id"], name: "index_feedbacks_on_user_id"
+  end
+
```
Comment (parndt) [inline db/schema.rb:144]:
```suggestion
```

File: db/schema.rb [hunk 2]
```diff
@@ -471,7 +481,7 @@
     t.integer "updated_by_id"
     t.boolean "allow_password_change", default: true
     t.integer "created_by_id"
-    t.datetime "archived_at"
+    t.boolean "is_archived", default: false, null: false
```
Comment (parndt) [inline db/schema.rb:484]:
```suggestion
```

File: db/schema.rb [hunk 3]
```diff
@@ -498,6 +508,7 @@
   add_foreign_key "actors", "actortypes"
   add_foreign_key "actortype_taxonomies", "actortypes"
   add_foreign_key "actortype_taxonomies", "taxonomies"
+  add_foreign_key "feedbacks", "users"
```
Comment (parndt) [inline db/schema.rb:511]:
```suggestion
```

**Comment by tmfrnz** [comment]
@parndt i am happy to confirm that this now works 🎉

the only question I have if it was possible to change the error message that is returned when archived users fail to login. currently the response is the same as for invalid login credentials (which is not entirely false but a bit misleading):
```
{"success":false,"errors":["Invalid login credentials. Please try again."]}
```

**Comment by parndt** [comment]
> the only question I have if it was possible to change the error message that is returned when archived users fail to login. currently the response is the same as for invalid login credentials (which is not entirely false but a bit misleading):

hmm.. maybe.. I'll look into it.

**Comment by tmfrnz** [comment]
> Invalid login credentials. Please try again.

thanks!

How about something like: "Your account has been archived."

**Review by parndt — APPROVED** [review]

**Pull request merged**